File: app/src/main/java/org/softeg/slartus/forpdaplus/listfragments/TopicsHistoryListFragment.java
package org.softeg.slartus.forpdaplus.listfragments;/*
 * Created by slinkin on 20.03.14.
 */

import android.view.ContextMenu;
import android.view.View;
import android.widget.AdapterView;

import org.softeg.slartus.forpdaapi.IListItem;
import org.softeg.slartus.forpdaapi.ListInfo;
import org.softeg.slartus.forpdaplus.Client;
import org.softeg.slartus.forpdaplus.R;
import org.softeg.slartus.forpdaplus.classes.MenuListDialog;
import org.softeg.slartus.forpdaplus.classes.common.ExtUrl;
import org.softeg.slartus.forpdaplus.common.AppLog;
import org.softeg.slartus.forpdaplus.db.TopicsHistoryTable;

import java.io.IOException;
import java.text.ParseException;
import java.util.ArrayList;
import java.util.List;

public class TopicsHistoryListFragment extends TopicsListFragment {
    public TopicsHistoryListFragment() {

        super();
    }

    @Override
    protected ArrayList<? extends IListItem> loadTopics(Client client, ListInfo listInfo) throws IOException, ParseException {
        return TopicsHistoryTable.getTopicsHistory(listInfo);
    }

    @Override
    public void onCreateContextMenu(ContextMenu menu, View v, ContextMenu.ContextMenuInfo menuInfo) {
        try {
            AdapterView.AdapterContextMenuInfo info = (AdapterView.AdapterContextMenuInfo) menuInfo;
            if (info.id == -1) return;
            Object o = getAdapter().getItem((int) info.id);
            if (o == null)
                return;
            final IListItem topic = (IListItem) o;

            final List<MenuListDialog> list = new ArrayList<>();
            list.add(new MenuListDialog(getString(R.string.delete_from_visited), new Runnable() {
                @Override
                public void run() {
                    TopicsHistoryTable.delete(topic.getId());
                    getMData().remove(topic);
                    getAdapter().notifyDataSetChanged();
                }
            }));
            ExtUrl.showContextDialog(getContext(), null, list);
        } catch (Throwable ex) {
            AppLog.e(ex);
        }

    }
}

File: app/src/main/java/org/softeg/slartus/forpdaplus/listfragments/TopicsHistoryListFragment.kt
package org.softeg.slartus.forpdaplus.listfragments

import android.view.ContextMenu
import android.view.ContextMenu.ContextMenuInfo
import android.view.View
import android.widget.AdapterView.AdapterContextMenuInfo
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import io.reactivex.schedulers.Schedulers
import org.softeg.slartus.forpdaapi.Forum
import org.softeg.slartus.forpdaapi.IListItem
import org.softeg.slartus.forpdaapi.ListInfo
import org.softeg.slartus.forpdacommon.sameContentWith
import org.softeg.slartus.forpdaplus.Client
import org.softeg.slartus.forpdaplus.MainActivity
import org.softeg.slartus.forpdaplus.R
import org.softeg.slartus.forpdaplus.classes.MenuListDialog
import org.softeg.slartus.forpdaplus.classes.common.ExtUrl
import org.softeg.slartus.forpdaplus.common.AppLog
import org.softeg.slartus.forpdaplus.db.TopicsHistoryTable
import org.softeg.slartus.forpdaplus.fragments.search.SearchSettingsDialogFragment
import org.softeg.slartus.forpdaplus.repositories.ForumsRepository
import java.io.IOException
import java.util.*

/*
 * Created by slinkin on 20.03.14.
 */
class TopicsHistoryListFragment : TopicsListFragment() {
    private var dataSubscriber: Disposable? = null
    private fun subscribesData() {
        dataSubscriber?.dispose()

        dataSubscriber =
            ForumsRepository.instance
                .forumsSubject
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(
                    {
                        loadData(true)
                    },
                    {
                        AppLog.e(activity, it)
                    }
                )
    }

    override fun onPause() {
        super.onPause()
        dataSubscriber?.dispose()
        MainActivity.searchSettings = SearchSettingsDialogFragment.createDefaultSearchSettings()
    }

    override fun onResume() {
        super.onResume()
        subscribesData()
    }

    @Throws(IOException::class)
    override fun loadTopics(client: Client?, listInfo: ListInfo?): ArrayList<out IListItem?> {
        return TopicsHistoryTable.getTopicsHistory(listInfo)
    }

    override fun onCreateContextMenu(menu: ContextMenu, v: View, menuInfo: ContextMenuInfo?) {
        try {
            val info = menuInfo as AdapterContextMenuInfo?
            if (info!!.id == -1L) return
            val o = adapter!!.getItem(info.id.toInt()) ?: return
            val topic = o as IListItem
            val list: MutableList<MenuListDialog> = ArrayList()
            list.add(MenuListDialog(getString(R.string.delete_from_visited)) {
                TopicsHistoryTable.delete(topic.id)
                mData.remove(topic)
                adapter!!.notifyDataSetChanged()
            })
            ExtUrl.showContextDialog(context, null, list)
        } catch (ex: Throwable) {
            AppLog.e(ex)
        }
    }
}